### PR TITLE
Update references and content after repo name change

### PR DESCRIPTION
**Pull request by dav3r**
#  #

## 🗣 Description ##

This PR updates all references in this repo from an old module name (`cisagov/new-user-alert-tf-module`) to the new module name (`cisagov/user-group-mod-alert-tf-module`).  This PR also includes some documentation changes to reflect the new functionality added in https://github.com/cisagov/user-group-mod-alert-tf-module/pull/25.

## 💭 Motivation and context ##

This PR is a direct result of the repo name change that followed https://github.com/cisagov/user-group-mod-alert-tf-module/pull/25.  

## 🧪 Testing ##

These changes were verified by successfully applying the Terraform in each workspace of each account subdirectory of this repo, including the large number of workspaces in the `dynamic` subdirectory.  FYI, I confirmed all of the updated SNS subscriptions as part of this process (i.e. I clicked on each "confirm" link in the AWS emails that were generated when this Terraform was applied).

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] All new and existing tests pass.

### CODE REVIEW

**Review by jasonodoom — APPROVED** [review]

File: audit/README.md [hunk 1]
```diff
@@ -78,8 +78,8 @@ future changes by simply running `terraform apply
 | Name | Source | Version |
 |------|--------|---------|
 | cw\_alarm\_sns | github.com/cisagov/sns-send-to-account-email-tf-module | n/a |
-| new\_user\_event | github.com/cisagov/new-user-alert-tf-module | n/a |
-| new\_user\_sns | github.com/cisagov/sns-send-to-account-email-tf-module | n/a |
+| user\_group\_mod\_event | github.com/cisagov/user-group-mod-alert-tf-module | n/a |
+| user\_group\_mod\_sns | github.com/cisagov/sns-send-to-account-email-tf-module | n/a |
 | provisionaccount | github.com/cisagov/provisionaccount-role-tf-module | n/a |
```
Comment (jsf9k) [inline audit/README.md]: These should be alphabetized.  The same goes for the other `README.md` files in this repo.

**Review by jsf9k — CHANGES_REQUESTED** [review]

File: audit/README.md (hunk 1 above)
Comment (dav3r) [reply audit/README.md]: Good catch!  Please see https://github.com/cisagov/cool-accounts/pull/140/commits/0ce4e5182517af04d781184bb95365532257b5ba.

Look for an upcoming PR shortly to add our standard `.terraform-docs.yml` file to each subdirectory, something that we overlooked when https://github.com/cisagov/skeleton-tf-module/pull/189 trickled down to this repo.

Comment (dav3r) [reply audit/README.md]: FYI, PR #141 has been created.

Comment (jsf9k) [reply audit/README.md]: Thanks @dav3r!

**Review by jsf9k — APPROVED** [review]

**Pull request merged**